src/napari_imagej/widget_IJ2.py

```diff
@@ -0,0 +1,124 @@
+from enum import Enum
+from typing import List
+
+from magicgui.widgets import request_values
+from napari import Viewer
+from napari.layers import Layer
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QIcon, QPixmap
+from qtpy.QtWidgets import QMessageBox, QPushButton, QVBoxLayout, QWidget
+
+from napari_imagej._module_utils import _get_layers_hack
+from napari_imagej.setup_imagej import ij, jc, running_headless
+
+
+class GUIWidget(QWidget):
+    def __init__(self, viewer: Viewer):
+        super().__init__()
+        self.setLayout(QVBoxLayout())
+
+        self.gui_button: GUIButton = GUIButton()
+        self.layout().addWidget(self.gui_button)
+
+        self.to_ij: ToIJButton = ToIJButton()
+        self.layout().addWidget(self.to_ij)
+
+        self.from_ij: FromIJButton = FromIJButton(viewer)
+        self.layout().addWidget(self.from_ij)
+        if not running_headless():
+            self.gui_button.clicked.connect(lambda: self.to_ij.setHidden(False))
+            self.gui_button.clicked.connect(lambda: self.from_ij.setHidden(False))
+
+
+class ToIJButton(QPushButton):
+    def __init__(self):
+        super().__init__()
+        self.setHidden(True)
+        self.setText("Send layers to ImageJ2")
+        self.clicked.connect(self.send_layers)
+
+    def send_layers(self):
+        layers: dict = request_values(
+            title="Send layers to ImageJ2",
+            layers={"annotation": Layer, "options": {"choices": _get_layers_hack}},
+        )
+        if layers is not None:
+            for _, layer in layers.items():
+                if isinstance(layer, Layer):
+                    name = layer.name
+                    data = ij().py.to_java(layer.data)
+                    ij().ui().show(name, data)
+
+
+class FromIJButton(QPushButton):
+    def __init__(self, viewer: Viewer):
+        super().__init__()
+        self.viewer = viewer
+
+        self.setHidden(True)
+        self.setText("Get layers from ImageJ2")
+        self.clicked.connect(self.get_layers)
+
+    def _get_objects(self, t):
+        compatibleInputs = ij().convert().getCompatibleInputs(t)
+        compatibleInputs.addAll(ij().object().getObjects(t))
+        return list(compatibleInputs)
+
+    def get_layers(self) -> List[Layer]:
+        images = self._get_objects(jc.RandomAccessibleInterval)
```

Comment:
Can we just encapsulate this argument within the function?

Reply:
Idk, I don't see why we'd do this anymore. Might be worth moving up to `module_utils` sometime...